src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -198,20 +198,33 @@ public void Start()
 
                 // STEP 2: Open the socket connection to the remote host
                 var cts = new CancellationTokenSource();
-                cts.CancelAfter(ConnectTimeout);
-                try {
+                var tok = cts.Token;
+                
+                tok.Register(() =>
+                {
+                    _client.Close();
```

Comment:
`tok.Register` returns a value that needs to be disposed here, and the `CancellationTokenSource` itself should also be disposed in here.